prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/model/nexus/impl/NexusImpl.java

```diff
@@ -64,9 +76,83 @@ public NexusImpl(ResourceSet resourceSet, SensiNactPackage sensinactPackage,
         this.resourceSet = resourceSet;
         this.sensinactPackage = sensinactPackage;
         this.notificationAccumulator = accumulator;
+        // TODO we need a general Working Directory for such data
+        Optional<EPackage> packageOptional = loadDefaultPackage(BASIC_BASE_ECORE);
 
-        defaultPackage = EMFUtil.createPackage("base", DEFAULT_URI, "sensinactBase", this.resourceSet);
+        defaultPackage = packageOptional
+                .orElseGet(() -> EMFUtil.createPackage("base", DEFAULT_URI, "sensinactBase", this.resourceSet));
         packageCache.put(DEFAULT_URI_OBJECT, defaultPackage);
+        loadInstances();
+    }
+
+    private Optional<EPackage> loadDefaultPackage(String fileName) {
+        Resource resource = resourceSet.createResource(URI.createFileURI(fileName));
+        if (new File(fileName).exists()) {
+            try {
+                resource.load(null);
+                if (!resource.getContents().isEmpty()) {
+                    EPackage defaultPackage = (EPackage) resource.getContents().get(0);
+                    resource.setURI(URI.createURI(defaultPackage.getNsURI()));
+                    return Optional.of(defaultPackage);
+                }
+            } catch (IOException e) {
+                Exceptions.duck(e);
```

Comment:
Ducking exceptions is (in my opinion) bad practice and we shouldn't do it. It will also allow us to remove the bnd libg dependency.

Another positive benefit will be that the constructor has to make a call about whether a failure to load the default package should blow up the creation of the NexusImpl or not (it does at the moment) and it will force some logging.

Reply:
Makes sense. The Main question would be, what we should do. If the load of the old Data fails for whatever reason, should we just log it as an error and keep on going?

Reply:
> should we just log it as an error and keep on going?

Initially yes. I think that it may eventually depend on the error type (e.g. a versioning error might indicate a failed migration in upgrade), but if the file is garbage then we probably want a big warning in the log and to continue without it.